memorystore/valkey/caching/GUIDE.md

```diff
@@ -56,120 +55,151 @@ Next, ensure the following dependencies have been added to your POM.xml file.
+    return itemId;
 }
 ```
 
-### Reading from the Cache (Retrieving Values)
+#### Reading from the Cache (Retrieving Values)
 
-This method demonstrates how to efficiently retrieve data using a caching layer (Memorystore) to improve performance. The function first checks the cache for the requested item and handles its time-to-live (TTL) appropriately. If the item is not found in the cache, it fetches the data from the database, caches it, and then returns the result.
+This method demonstrates how to efficiently retrieve data using a caching layer (Memorystore) to improve performance.
+
+**Step 1**: This function searches the cache to see if an item exists, if found the cached itme is returned.
+**Step2**: If the item is not found in the cache, it is retrieved from the database. If no record exists, then `null` is returned.
+**Step3**: The database item is then turned into a string and cached in the datbase with the default TTL.
+**Step4**: The database item is then returned.
 
 ```java
 /** Import the Jedis library */
 import redis.clients.jedis.Jedis;
 
+/** Set a default value for Time-to-live(TTL) **/
+public static final Long DEFAULT_TTL = 60 L;
+
 public Item get(long id) {
-   /** Ensure that the item id is a string for retirval from Memorystore */
+    // Ensure that the item id is a string for retrieval from Memorystore
     String idString = Long.toString(id);
 
-    /* Check if the item exists in  the cache */
-    if (jedis.exists(idString)) {
-      // If the data exists in the cache extend the TTL
-      jedis.expire(idString, DEFAULT_TTL);
+    try {
+        // Attempt to get the cached item from Memorystore
+        String cachedValue = jedis.get(idString);
+
+        // Check if we have found a valid cache item
+        if (cachedValue != null) {
+            // Set a property to display cached item as a property for usage in the application
+            cachedItem.setFromCache(true);
 
-      // Return the cached data
-      Item cachedItem = Item.fromJSONString(jedis.get(idString));
-      cachedItem.setFromCache(true);
-      return cachedItem;
+            // Extract the item into a data object
+            Item cachedItem = Item.fromJSONString(cachedValue);
+
+            /** Return the cached item **/
+            return cachedItem;
+        }
+    } catch (Exception e) {
+        // If there's an error with the cache, log the error and continue
+        System.err.println("Error with cache: " + e.getMessage());
     }
 
-   /** Search for the item in the database */
-   Optional<Item> item = itemsRepository.get(id);
+    // No cached item exists, search for the item in the database
+    Optional < Item > item = itemsRepository.get(id);
 
-    /** Check if a record has been found, If the data doesn't exist in the database, return null */
+    // Check if a record has been found, If the data doesn't exist in the database, return null
     if (item.isEmpty()) {
-      return null;
+        return null;
```

Comment:
curious if it's possible to cache a null value.

There's pros & cons to this. in production if you're used to a high cache hit ratio & client keeps retrying on a missing item you'll see a lot of traffic go to the database b/c the cache is not protecting.

sometimes folks will put the null in cache but with a shorter TTL.

Reply:
I don't believe it possible to cache null values.

An example error message would be 

```bash
Setex with null value failed as expected: Cannot read the array length because "raw" is null
```

A potential alternative would be set an empty string or a placeholder value `<Empty>`. An update on any front-end applications would be required to know how to handle the returned value.

Reply:
sgtm thanks for looking.